chain/ethereum/src/runtime/runtime_adapter.rs

```diff
@@ -45,9 +45,10 @@ impl blockchain::RuntimeAdapter<Chain> for RuntimeAdapter {
         let call_cache = self.call_cache.cheap_clone();
         let eth_adapter = self
             .eth_adapters
-            .cheapest_with(&NodeCapabilities {
+            .cheapest_with_for_call(&NodeCapabilities {
```

Comment:
This is the only place where we'd use a call only adapter, so that's why I thought it could use a dedicated method that doesn't need a `NodeCapabilities` argument.